Dockerfile

```diff
@@ -89,7 +89,7 @@ RUN mkdir /.cache /.local
 ADD runtimes/python3-9 /workspace
 RUN chown -R 9653 /.cache /.local /workspace
 WORKDIR /workspace
-RUN apk add git
+RUN apk add git gcc musl-dev
```

Comment:
these are some serious dependencies - and we're trying for minimal dependencies on the runtimes - what's the need for them?

Reply:
I modified this docker image to not use `git` dependency at all, by moving python dsl & sdk sources into that docker image and building them directly in the container. 

One of the changes I did in this PR, is modifying the SDK to support asyncio handlers, which also helped with implementing Generator Step handlers. Implementing it in asyncio fashion is I think cleaner than doing it with threads. But that meant that I need a dependency `aiohttp`. This dependency requires `gcc` & `musl-dev` to correctly build when installing it via pip. So in the Dockerfile, I specified that those two should be installed right before installing deps for the sdk, and then removed once sdk deps are built. I tested it and it seems to be working.

What do you think about such an approach to keeping the images small?

Reply:
This also speeds up `python3-9` runtime docker image startup time, as now it doesn't need to download dependencies on startup. Instead, those are already included in the image.